### DISCUSSION TITLE
"Raw Go" feature

### DISCUSSION THREAD

**Discussion by maxodoble** [a-h/templ#787] [General]
Hello,

i am very tempted to use the experimental "Raw Go" feature [(see Docs)](https://templ.guide/syntax-and-usage/raw-go) for var declarations inside of templates to avoid multiple lengthy function calls.

BUT: the caution statement in the docs ("This page describes functionality that is experimental, not enabled by default, and may change or be removed in future versions.") is kind of telling me: don't use this feature.

so i'd be interested in the opinion of the community regarding this feature:

- use it?
- avoid it?

Thanks,
Max

**Comment by joerdav** [comment]
Hey! Thanks for reaching out.

Experimental features are there to be tested, so if you are happy to take on the risk that it may change in a future version, or may not be fully complete then you can contribute to templ by trying it out and feeding back, and helping it became a fully-fledged feature.

That being said, the raw go feature was highly demanded and does solve a problem, so I would say that the likelihood of removal is low. @a-h may be able to back me up or let me know I'm talking out of line there!

I would say that the main reason it is feature flagged is because it is quite a wide reaching change, so it seemed safer to feature flag whilst we ironed out and surfaced any issues (there have already been some that were raised and fixed), rather than it being an experiment to decide whether we want the feature or not (I think we do want it).

Of course I'm a maintainer so that doesn't really answer your question on the community opinion!

**Reply by maxodoble** [reply]
thank you for your thorough reply, this is much appreciated!

Max

**Reply by a-h** [reply]
Yes, agree with @joerdav - I think the feature is here to stay.

So far, people haven't done anything weird with it, or pointed out any significant flaws. I was worried about people trying to do things like do `for` loops that render components inside there, and then getting confused.

**Reply by matt9mg** [reply]
I also agree with this, the rawgo feature is really helpful especially when rendering Flash Messages.

I hope this becomes a permanent feature I'm using this all over the shop and it works extremely well

Thanks for all the effort! A great templating solution 🫶

**Reply by emehrkay** [reply]
It's good to know that rawgo will stick around. 

> Im about to get weird with it though @a-h, something like `{{ fmt.Println("hello") }} {{ var x = func() {fmt.Println("world")}}} {{ x() }}` in the middle of a template. sorry

But seriously, rawgo makes templ a lot more useful than it already is.

**Reply by nexovec** [reply]
Finally I don't have to write a go wrapper for every nontrivial template. Thank you guys.

**Reply by Napam** [reply]
+1 for this feature. It would let me write way less boilerplate functions as I like to dynamically change classes and whatnot based on some logic.

EDIT: Just saw that it was released four days ago 🎉. The docs are still saying it is experimental.

**Reply by a-h** [reply]
Ah, thanks @Napam - I'd forgotten to run a docs push, I've triggered one: https://github.com/a-h/templ/actions/workflows/docs.yaml
